apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -336,6 +336,18 @@ type EnvoyListenerConfig struct {
 	// +optional
 	DisableMergeSlashes *bool `json:"disableMergeSlashes,omitempty"`
 
+	// Defines the action to be applied to the Server header on the response path
+	// When configured as OVERWRITE, Overwrites any Server header with the contents of server_name.
+	// When configured as APPEND_IF_ABSENT, ⁣If no Server header is present, append Server server_name If a Server header is present, pass it through.
+	// When configured as PASS_THROUGH, ⁣Pass through the value of the server header, and do not append a header if none is present.
+	//
+	// Values: `OVERWRITE` (default), `APPEND_IF_ABSENT`, `PASS_THROUGH`
```

Comment:
@sunjayBhatia WDYT about casing here of the values here? For DNS lookup family we lowercased, but I don't see existing multi-word options. I guess we could switch these to lowercase, but honestly not sure if there's any point in doing so.

Reply:
actually it's an integer in envoy

Reply:
hm yeah as you say many if not all the other relevant `const`s in this package and `pkg/config` we define are lowercase, so for consistency we could switch to that, since we're not really going to pass the value straight through to Envoy it doesn't really matter